packages/ui/src/components/modal/index.ts

```diff
@@ -1 +1,8 @@
-export { default } from "./modal.tsx";
+export { 
+  Modal, 
+  ModalClose, 
+  ModalContent, 
+  ModalDescription,
+  ModalTitle, 
+  ModalTrigger 
```

Comment:
Do you mean abstracting where developers just import a 
```tsx 
<Modal title='hi'> 
   <p> how are you </p>
 <Modal/>
 ```
 
 Not sure if we are on the same page. But I think there are 3 schools of thoughts for composing modal. 
#### Compositionally flexible like above
- [ Material-UI](https://mui.com/material-ui/react-dialog/)
- [Charkra-UI](https://chakra-ui.com/docs/components/alert-dialog/usage)
#### Hybrid
- [Blue print](https://blueprintjs.com/docs/#core/components/dialog)
#### Limited 
- [Ant Design](https://ant.design/components/modal#modal-demo-width)


If we choose to go with the abstracting to a single import. We will be dealing with things this like below. Personally. I don't like nesting entire html stuff into props. React Native had alot of this, and it was really hard to read for me. 
```tsx

    footer={[
          <Button key="back" onClick={handleCancel}>
            Return
          </Button>,
          <Button key="submit" type="primary" loading={loading} onClick={handleOk}>
            Submit
          </Button>,
          <Button
            key="link"
            href="https://google.com"
            type="primary"
            loading={loading}
            onClick={handleOk}
          >
            Search on Google
          </Button>,
        ]}
      >
```

and too add styling to description it may look like
 
 ```tsx
 <Modal 
      title='hi' 
      description={
        <p>className="mb-4 text-base leading-6">Parking Insights lets you get 
        <span className='font-semibold'>data on Los Angeles parking tickets</span> 
        data on Los Angeles parking tickets using several options:</p>
>
     <p> content </p>
</Modal> 
 ```